# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTypes.cpp
# Commit: b620533c6d63
# Full Hash: b620533c6d633ef50c5e5cb3ecd0f398206eb271
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-06-17 21:52:06
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Refactor wasm::Frame to reduce number of gnarly casts. r=lth,wingo
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79072
# ==============================================================================

diff -r f7ec4a920540 -r b620533c6d63 js/src/wasm/WasmTypes.cpp
--- a/js/src/wasm/WasmTypes.cpp	Wed Jun 17 11:41:13 2020 +0000
+++ b/js/src/wasm/WasmTypes.cpp	Wed Jun 17 12:04:54 2020 +0000
@@ -634,7 +634,7 @@
 
 /* static */
 DebugFrame* DebugFrame::from(Frame* fp) {
-  MOZ_ASSERT(fp->tls->instance->code().metadata().debugEnabled);
+  MOZ_ASSERT(fp->instance()->code().metadata().debugEnabled);
   auto* df =
       reinterpret_cast<DebugFrame*>((uint8_t*)fp - DebugFrame::offsetOfFrame());
   MOZ_ASSERT(fp->instance() == df->instance());